structured_experiments/2022.04.28.notebook_template/notebooks/Template - Kaggle Sample .ipynb

```diff
@@ -0,0 +1,186 @@
+    "tesstrain.train_tesseract(\"/train/tesstrain\",  model_name=model_name, start_model=\"eng\", tessdata=tessdata, max_iterations=100)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "3b3129fb",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['/train/tesstrain/data/storysquad-ground-truth/31-3117-1-1.png', '/train/tesstrain/data/storysquad-ground-truth/31-3112-1-14-1.png'] ['gum drop land', 'The end']\n"
+     ]
+    }
+   ],
+   "source": [
+    "# get a random sample of images from TEST_PATH folder and read their ground truth files (labels with correct OCR)\n",
+    "test_image_paths = []\n",
+    "test_image_labels = []\n",
+    "count = 0\n",
+    "\n",
+    "for file_name in listdir(TEST_PATH):\n",
+    "    file_path = join(TEST_PATH, file_name)\n",
+    "    if count < 100 and file_path.endswith(\".png\") and isfile(file_path):\n",
+    "        \n",
+    "        ground_truth_path = file_path.replace(\".png\", \".gt.txt\")\n",
+    "        if isfile(file_path) and isfile(ground_truth_path):\n",
+    "            test_image_paths.append(file_path)\n",
+    "            with open(ground_truth_path, mode='r') as f:\n",
+    "                ground_truth = f.read()\n",
+    "            test_image_labels.append(ground_truth)\n",
+    "            count += 1\n",
+    "    else:\n",
+    "        continue\n",
+    "print(test_image_paths[:2], test_image_labels[:2])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "6a22132f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# generate word error rate and character error rate for model based on random sample generated above\n",
+    "# see ./structured_experiments/2022.04.28.notebook_template/mytesseract/tesstrain/__init__.py for implementation\n",
+    "word_error_rate, char_error_rate = tesstest.test_model(\"kaggle\", tessdata, test_image_paths, test_image_labels)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "74b0d708",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Word Error Rate of 100.0% and Character Error Rate of 92.87%\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Word Error Rate of {round(100 *word_error_rate, 2)}% and Character Error Rate of {round(100 *char_error_rate, 2)}%\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3 (ipykernel)",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.13"
+  }
+ },
+ "nbformat": 4,
```

Comment:
what does "nbformat": 4 affects?

Reply:
This is how GitHub renders the notebook diffs, if you view the file though it will show you a more human readable version of what's in the notebook. I wouldn't really worry about this one 👍